### PR TITLE
Try harder to get a good log source class in EventSourcedBehavior

### LINKED ISSUE #26564: Surprising log source with event sourced behavior
@2m showed me this the other day: https://github.com/2m/akka/commit/20fa1f8e2e8b4593dbab7215db768b3b41d1004a

> the one that logs in the Behavior.setup is fine, but all of the other ones, either from command/evend handler or signal handlers, they have the internal event sourced bahvior class name attached

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #26564

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM after cleanup

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/TimerSpec.scala [hunk 1]
```diff
@@ -336,5 +336,29 @@ class TimerSpec extends ScalaTestWithActorTestKit("""
...
+        timered
+    }
+    "böö bööö böö" in {
+
+      val ref = spawn(non)
+
+      ref ! "switch"
+      ref ! "switch"
+      Thread.sleep(3000)
+    }
```
[the remark this answers is not in the document]
Comment (johanandren) [reply akka-actor-typed-tests/src/test/scala/akka/actor/typed/TimerSpec.scala]: Ups, no it was trying to reproduce another issue, I'll get rid of it.

File: akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/LoggerSourceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    // note that these are somewhat intermingled to make sure no log event from
+    // one test case leaks to another, the actual log class is what is tested in each individual case
+
+    "log from setup" in {
+      EventFilter.info("recovery-completed", occurrences = 1).intercept {
+        eventFilterFor("setting-up-behavior").intercept {
+          spawn(behavior)
+        }
+      }
+
```
Comment (helena) [inline akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/LoggerSourceSpec.scala]: Nit: no space needed

**Review by helena — APPROVED** [review]
Looks great!

**Pull request merged**